src/relevance_feedback/relevance_feedback.py

```diff
@@ -0,0 +1,81 @@
+import cv2, os
+import numpy as np
+from scipy.stats import moment
+from itertools import repeat
+import pickle
+
+import os,sys,inspect
+sys.path.insert(0, '../backend/')
+import singular_value_decomposition
+from database_connection import DatabaseConnection
+from utils import get_most_m_similar_images
+
+class RelevanceFeedback:
+
+	def __init__(self):
+		self.database_connection = DatabaseConnection()
+		self.conn = self.database_connection.get_db_connection()
+		print('Initiating RelevanceFeedback')
+
+	def compute_new_query_vector(self,q_old,relevant_items,irrel_items,alpha=0.5,beta=0.45,gamma=0.05):
+		print('Computing new query vector.....')
+		
+		avg_rel_vec=np.zeros(q_old.shape)
+		avg_irl_vec=np.zeros(q_old.shape)
+
+		#Aggregating relevant items
+		for item in relevant_items:
+			vector=self.database_connection.get_feature_data_for_image('histogram_of_gradients',item)
+			avg_rel_vec=avg_rel_vec+vector
+
+		#Aggregating irrelevant items
+		for item in irrel_items:
+			vector=self.database_connection.get_feature_data_for_image('histogram_of_gradients',item)
+			avg_irl_vec=avg_irl_vec+vector
+
+		avg_rel_vec=avg_rel_vec/len(relevant_items)
+		avg_irl_vec=avg_irl_vec/len(irrel_items)		
+
+		q_new= alpha*q_old + beta*avg_rel_vec - gamma*avg_irl_vec
+		return q_new
+
+	def get_user_feedback(self,init_rank_list,q_name):
```

Comment:
We are going to ask for relevance only once? I think we will be calling it multiple times until the user is satisfied

Reply:
We can always reuse this function in a loop... but i think it will be once